extensions/native/compiler/src/constraints/halo2/baby_bear.rs

```diff
@@ -351,28 +352,61 @@ impl BabyBearChip {
     }
 }
 
-/// Constrains and returns `(c, r)` such that `a = b * c + r`.
+/// Constrains and returns `(c, r)` such that `a = BabyBear::ORDER_U32 * c + r`.
 ///
 /// * a: [QuantumCell] value to divide
-/// * b: [BigUint] value to divide by
 /// * a_num_bits: number of bits needed to represent the absolute value of `a`
 ///
 /// ## Assumptions
-/// * `b != 0` and that `abs(a) < 2^a_max_bits`
 /// * `a_max_bits < F::CAPACITY = F::NUM_BITS - RESERVED_HIGH_BITS`
 ///   * Unsafe behavior if `a_max_bits >= F::CAPACITY`
 fn signed_div_mod<F>(
     range: &RangeChip<F>,
     ctx: &mut Context<F>,
     a: impl Into<QuantumCell<F>>,
-    b: impl Into<BigUint>,
```

Comment:
the function still works in greater generality, I guess you wanted it fixed for BabyBear to simplify the logic / explanation?